api/v1/vouchers/models.py

```diff
@@ -0,0 +1,59 @@
+import random
+from datetime import datetime
+from flask.ext.sqlalchemy import SQLAlchemy
+
+from stdnum import verhoeff
+
+from ..exceptions import VoucherException
+db = SQLAlchemy()
+
+__author__ = 'kenneth'
+
+
+class Voucher(db.Model):
+    __tablename__ = 'voucher_vouchers'
+
+    id = db.Column(db.Integer, primary_key=True)
+    code = db.Column(db.String(6))
+    redeemed_on = db.Column(db.DateTime(timezone=True), nullable=True)
+    created_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now())
+    modified_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now(), server_onupdate=db.func.now())
+    redeemed_by = db.Column(db.String(13), nullable=True)
```

Comment:
I am sure it defaults to UCT, which is OK or otherwise it will have to
stamp on the server timezome. I think org and flow makes sense, I'll be
able to add those.

On Wed, Jun 10, 2015 at 6:21 PM, Evan Wheeler <notifications@github.com>
wrote:

> In api/v1/vouchers/models.py
> <https://github.com/ewheeler/rapidpro-webhooks/pull/1#discussion_r32129101>
> :
>
> > +
> > +from ..exceptions import VoucherException
> > +db = SQLAlchemy()
> > +
> > +__author__ = 'kenneth'
> > +
> > +
> > +class Voucher(db.Model):
> > +    __tablename__ = 'voucher_vouchers'
> > +
> > +    id = db.Column(db.Integer, primary_key=True)
> > +    code = db.Column(db.String(6))
> > +    redeemed_on = db.Column(db.DateTime(timezone=True), nullable=True)
> > +    created_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now())
> > +    modified_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now(), server_onupdate=db.func.now())
> > +    redeemed_by = db.Column(db.String(13), nullable=True)
>
> do we get the flow id and/or the org id from rapidpro? might be good to
> store those too so that if multiple orgs/countries are using this voucher
> endpoint then we're able to distinguish who is redeeming ... this would
> also mean that we'd want to associated the vouchers to a certain org..
> what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ewheeler/rapidpro-webhooks/pull/1/files#r32129101>.
>